packages/language/src/preprocessor/instructions.ts

```diff
@@ -156,12 +156,29 @@ export interface NoteInstruction {
   code: ExpressionInstruction;
 }
 
-export interface IfInstruction {
-  kind: InstructionKind.If;
-  element: ast.IfStatement;
-  condition: ExpressionInstruction;
-  trueBranch?: InstructionNode;
-  falseBranch?: InstructionNode;
+export interface SelectInstruction {
+  kind: InstructionKind.Select;
+  element: ast.SyntaxNode;
+  compare?: ExpressionInstruction;
+  cases: Cases[];
+}
+
+export function createSelectInstruction(
+  element: ast.SyntaxNode,
+  compare: ExpressionInstruction | undefined,
+  cases: Cases[],
+): SelectInstruction {
+  return {
+    kind: InstructionKind.Select,
+    element,
+    compare,
+    cases,
+  };
+}
+
+export interface Cases {
+  conditions: ExpressionInstruction[];
```

Comment:
Empty conditions === OTHERWISE?
May add a JS Doc comment with this hint to this field.